File: tests/nla/Freire1.java
public class Freire1 {
    public static void vtrace1(float a, float x, float r){}
    public static void vtrace2(float a, float x, float r){}
    
    public static void main (String[] args) {
    }

    public static void mainQ(float a) {
	float x = a/2.0f;
	float r = 0;

	while(true){
	    vtrace1(a,x,r);
	    //assert((float)a == 2*x + r*r - r); 
	    if (!(x>r)) break;
	    x = x - r;
	    r = r + 1;
	}

	vtrace2(a,x,r);
	//assert((float)a == 2*x + r*r - r); 
	//return r;	  
    }
}

File: old/dig2_darpa2/programs/nla/freire1.c
#include <stdio.h>
#include <assert.h>

int mainQ(int a){
     double x = ((double)a)/2.0;
     int r = 0;


     while(1){
	  //assert((double)a == 2*x + r*r - r); 
	  //%%%traces: int a, double x, int r
	  
	  if (!(x>r)) break;
	  x = x - r;
	  r = r + 1;
     }

     //assert(r==(int)round(sqrt(a)));
     return r;
}


int main(int argc, char **argv){
     mainQ(atoi(argv[1]));
     return 0;
}

